src/clj/rems/application/commands.clj

```diff
@@ -320,6 +321,13 @@
        (mapv (fn [license]
                {:license/id (:id license)}))))
 
+(defn- entitlement-end-not-in-future-error
+  "Checks that entitlement date time, if provided, is in the future."
+  [dt]
+  (when dt
+    (when-not (time/after? dt (time/today-at 23 59 59))
+      {:errors [{:type :t.actions.errors/entitlement-end-not-in-future}]})))
```

Comment:
Sometimes we add the value e.g. `:entitlement-end "xyz"` to the error map to have an easier time debugging or API error message. We don't always show these in the UI.